CONTRIBUTING.md

```diff
@@ -0,0 +1,74 @@
+
+Generally, pull requests should consist of a single logical commit. However, if your PR is for a large feature, you may need a more logical breakdown of commits. This is fine as long as each commit is a single logical unit. Changes generated by `go mod tidy` or `go generate` should go in separate commits.
+
+Pull requests should target the release/vX.Y branch instead of master. The latest code in RKE is included in such release branches.
+
+### Writing good commit messages
+As the change is already well described in the GitHub issue and the PR, Git commit messages can be short or even empty, as long as the title of the commit can easily summarize what the change.
+
+### Reviewing, addressing feedback, and merging
+Generally, pull requests need two approvals from maintainers to be merged.
+
+When addressing review feedback, it is helpful to the reviewer if additional changes are made in new commits. This allows the reviewer to easily see the delta between what they previously reviewed and the changes you added to address their feedback. Once positively reviewed, those commits should be squashed into one.
+
+Once a PR has at least two approvals, it can be merged. Here’s how the merge should be handled:
+- If the PR is a single logical commit, the merger should use the “Rebase and merge” option. This keeps the git commit history very clean and simple and eliminates noise from "merge commits."
+- If the PR is more than one logical commit, the merger should use the “Create a merge commit” option.
+- If the PR consists of more than one commit because the author added commits to address feedback, the commits should be squashed into a single commit (or more than one logical commit, if it is a big feature that needs more commits). Remember that `go mod tidy` or `go generate` should go in separate commits, hence do not swaush in that case. Squashing of commits can be achieved in one of two ways:
+  - The merger can use the “Squash and merge” option. If they do this, the merger is responsible for cleaning up the commit message according to the previously stated commit message guidance.
+  - The pull request author, after getting the requisite approvals, can reorganize the commits as they see fit (using, for example, git rebase -i) and re-push.
+
+Once the PR is merged, please change the status of the linked issue in [rancher project](https://github.com/rancher/rancher) to "Needs QA". In the following days, the QA team will run tests and process the PR to verify that no regressions were introduced. Please be alert on the issue as the QA team might ask questions related to the PR. 
```

Comment:
Thought we also need to move the issue to `To-Test`, is that no longer applicable?